echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/orca/PipelineInitiator.java

```diff
@@ -150,6 +169,33 @@ private void onOrcaError(Pipeline pipeline, Throwable error) {
     log.error("Error triggering pipeline: {}", pipeline, error);
   }
 
+  /**
+   * @param user A service account name (or 'anonymous' if not specified)
+   * @return the allowed accounts for {@param user} as determined by fiat
+   */
+  private Set<String> getAllowedAccountsForUser(String user) {
+    if (fiatPermissionEvaluator == null || !fiatStatus.isLegacyFallbackEnabled()) {
+      return Collections.emptySet();
+    }
+
+    UserPermission.View userPermission = null;
+    try {
+      userPermission = fiatPermissionEvaluator.getPermission(user);
+    } catch (Exception e) {
+      log.error("Unable to fetch permission for {}", user, e);
+    }
+
+    if (userPermission == null) {
+      return Collections.emptySet();
+    }
+
+    return userPermission.getAccounts()
+      .stream()
+      .filter(v -> v.getAuthorizations().contains(Authorization.WRITE))
```

Comment:
why are we filtering here?

Reply:
I'll toss a comment but this aligns with what `gate` does in [AllowedAccountsSupport](https://github.com/spinnaker/gate/blob/master/gate-core/src/main/groovy/com/netflix/spinnaker/gate/security/AllowedAccountsSupport.java).